packages/e2e-test-utils/src/customFonts.js

```diff
@@ -13,11 +13,60 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+function parseText(optionsText) {
+  return optionsText.map((option) => {
+    const font = option.split('\n');
+    return { name: font[0], url: font[1] };
+  });
+}
+
+/**
+ * @typedef {Object} FontData font data parsed from custom font listbox
+ * @property {string} name font family name.
+ * @property {string} url url for font.
+ */
 
 /**
- * Internal dependencies
+ * Get array of fonts that have been added
+ *
+ * @return {Promise<[FontData]>} Fonts list.
  */
-import visitSettings from './visitSettings';
+export const getFontList = async () => {
+  try {
+    const optionsText = await page.evaluate(() =>
+      Array.from(
+        document.querySelectorAll(
+          'div[role=listbox] [role=option] div:first-child'
+        ),
+        (element) => element.innerText
+      )
+    );
+    return parseText(optionsText);
+  } catch (e) {
+    return [];
+  }
+};
+
+/**
+ * Get currently selected font in the listbox
+ *
+ * @return {Promise<FontData>} selected font data.
+ */
+export const getSelectedFont = async () => {
+  try {
+    const optionsText = await page.evaluate(() =>
+      Array.from(
+        document.querySelectorAll(
+          'div[role=listbox] [aria-selected=true] div:first-child'
+        ),
+        (element) => element.innerText
+      )
```

Comment:
Why are querying all the selectors here. Why can't we just do `document.querySelector` instead of `document. querySelectorAll`

Reply:
Updated to work better for working with single element `document.querySelector`